responses/test_responses.py

```diff
@@ -1177,6 +1177,43 @@ def run():
     assert_reset()
 
 
+def test_content_length_error():
+    """
+    Currently 'requests' do not enforce content length validation,
+    (validation that body length matches header). However, this could
+    be expected in next major version, see
+    https://github.com/psf/requests/pull/3563
+
+    Now user can manually patch URL3 lib to achieve the same
+    """
+
+    @responses.activate
+    def run():
+        responses.add(
+            responses.GET,
+            "http://example.com/api/123",
+            json={"message": "this body is too large"},
+            adding_headers={"content-length": "2"},
+        )
+        with pytest.raises(ChunkedEncodingError) as exc:
+            requests.get("http://example.com/api/123")
+
+        assert "IncompleteRead" in str(exc.value)
+
+    original_init = getattr(requests.packages.urllib3.HTTPResponse, "__init__")
+
+    def patched_init(self, *args, **kwargs):
+        original_init(self, *args, enforce_content_length=True, **kwargs)
+
+    setattr(requests.packages.urllib3.HTTPResponse, "__init__", patched_init)
```

Comment:
Should we be using mock.patch() instead of manually patching?

Reply:
@markstory 
`monkeypatch` should do the job. Still, not to go into recursion have to get attribute